### PR TITLE
refactor: improve memory and redirect managment

### PR DESCRIPTION
**Pull request by aleacevedo**
## Summary

Improve memory managment to save the less possible info. Improve redirect managment to avoid wrong redirection before the user execute the whole operation

### CODE REVIEW

File: src/modules/controller/information.js [hunk 1]
```diff
@@ -109,7 +109,7 @@ const InformationModule = (dispatch, getState) => {
 
   const isUnlock = {
     methodName: 'isUnlock',
-    handler: () => {},
+    handler: () => { },
```
Comment (ignaciosantise) [inline src/modules/controller/information.js:112]: thats good code

Comment (tomiir) [reply src/modules/controller/information.js:112]: 👏

**Review by ignaciosantise — APPROVED** [review]

File: src/screens/flows/WalletConnect/screens/Flows/index.tsx [hunk 2]
```diff
@@ -41,14 +41,17 @@ function WCFlows() {
...
+    requestId,
     metadata,
     args,
     handleApproveArgs,
     handleDeclineArgs,
     handleError,
     loading,
   } = (params || {}) as FlowsParams;
+  const request = useSelector(
+    (state: any) => state.walletconnect.pendingCallRequests[requestId]
```
Comment (mattgle) [inline src/screens/flows/WalletConnect/screens/Flows/index.tsx:53]: use the proper type instead of any

File: src/utils/deepLink.js [hunk 3]
```diff
@@ -11,14 +11,17 @@ import Navigation from '@/utils/navigation';
 
 function handleWalletConnect(uri, isUnlocked) {
   const { dispatch } = store;
-  dispatch(walletConnectSetPendingRedirect());
+  const { query } = new URL(uri);
+  const { bridge, requestId } = qs.parse(query.substring(1));
```
Comment (mattgle) [inline src/utils/deepLink.js:15]: does this `qs.parse(query.substring(1));` always has `{ bridge, requestId }`?

**Review by mattgle — COMMENTED** [review]
Just a few comments!

File: src/utils/deepLink.js (hunk 3 above)
Comment (aleacevedo) [reply src/utils/deepLink.js:15]: Not always but it is the idea, if  bridge is undefined it is not a firstConnection uri

**Pull request merged**